spec/jobs/dashboard_stats_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ::Jobs::DashboardStats do
+  let(:group_message) { GroupMessage.new(Group[:admins].name, :dashboard_problems, limit_once_per: 7.days.to_i) }
+
+  def clear_recently_sent!
+    Discourse.redis.del(group_message.sent_recently_key)
+  end
+
+  before do
+    clear_recently_sent!
+  end
+
+  it 'creates group message when problems are persistent for 2 days' do
+    Discourse.redis.setex(AdminDashboardData.problems_started_key, 14.days.to_i, Time.zone.now.to_s)
+    expect { described_class.new.execute({}) }.not_to change { Topic.count }
+
+    Discourse.redis.setex(AdminDashboardData.problems_started_key, 14.days.to_i, 3.days.ago)
+    expect { described_class.new.execute({}) }.to change { Topic.count }
+  end
+
+  it 'does not duplicate messages' do
+    Discourse.redis.setex(AdminDashboardData.problems_started_key, 14.days.to_i, 3.days.ago)
+    expect { described_class.new.execute({}) }.to change { Topic.count }
+    topic = Topic.last
+    clear_recently_sent!
+
+    expect { described_class.new.execute({}) }.not_to change { Topic.count }
+    expect(topic.reload.deleted_at).not_to be_nil
+  end
+
+  it 'duplicates message if previous one has replies' do
+    Discourse.redis.setex(AdminDashboardData.problems_started_key, 14.days.to_i, 3.days.ago)
+    expect { described_class.new.execute({}) }.to change { Topic.count }
+    clear_recently_sent!
+
+    _reply_1 = Fabricate(:post, topic: Topic.last)
+    expect { described_class.new.execute({}) }.to change { Topic.count }
+  end
+
+  it 'duplicates message if previous was 3 months ago' do
```

Comment:
I believe this should be deletes and not duplicated ?

Reply:
No, it was changed after Robin's comment https://github.com/discourse/discourse/pull/14319#discussion_r707480652 "Include a maximum date range. It's probably OK if they are warned again after a few months."

So my understanding was that if last message was long time ago, we can duplicate

Reply:
O in this case we're missing a test for deleting the duplicated message.

Reply:
Hm, no, this test is ensuring that we have 2 messages (3 months old and one recent). In that case, duplication is fine - we can have 2 messages

Another test is ensuring that if you have a message 2 weeks old, and new one, exactly same will be created - in that case, 2 weeks old should be deleted: https://github.com/discourse/discourse/pull/14319/files#diff-135d1a94ab5e9ffdc235d210cbcba2125fe5cff802cd9f0220daf81297cce350R24